satellite/metainfo/metainfo.go

```diff
@@ -847,3 +852,239 @@ func convertBucketToProto(ctx context.Context, bucket storj.Bucket) (pbBucket *p
 		},
 	}
 }
+
+// BeginObject begins object
+func (endpoint *Endpoint) BeginObject(ctx context.Context, req *pb.ObjectBeginRequest) (resp *pb.ObjectBeginResponse, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	keyInfo, err := endpoint.validateAuth(ctx, macaroon.Action{
+		Op:            macaroon.ActionWrite,
+		Bucket:        req.Bucket,
+		EncryptedPath: req.EncryptedPath,
+		Time:          time.Now(),
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Unauthenticated, err.Error())
+	}
+
+	err = endpoint.validateBucket(ctx, req.Bucket)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	bucket, err := endpoint.metainfo.GetBucket(ctx, req.Bucket, keyInfo.ProjectID)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	// take bucket RS values if not set in request
+	pbRS := req.RedundancyScheme
+	if pbRS.ErasureShareSize == 0 {
+		pbRS.ErasureShareSize = bucket.DefaultRedundancyScheme.ShareSize
+	}
+	if pbRS.MinReq == 0 {
```

Comment:
I wonder whether this check should be: `if pbRS.MinReq == 0 && pbRS.RepairThreshold == 0 ... { use default }` and then `if pbRS.MinReq == 0 || pbRS.RepairThreshold == 0 { return invalid arguments error }`. That way if you only partially supply the values it will fail, rather than default them to something. But we can figure out this in a later PR.

Reply:
yeah, I was not sure if we should take all or nothing. for now I follow approach from libuplink https://github.com/storj/storj/blob/ae8b9698f936f65972b83f2887a91bc1d93384ff/lib/uplink/bucket.go#L137